src/lib/crypto/hash.cpp

```diff
@@ -222,7 +222,8 @@ pgp_hash_add_int(pgp_hash_t *hash, unsigned n, size_t length)
 int
 pgp_hash_add(pgp_hash_t *hash, const void *buf, size_t len)
 {
-    return botan_hash_update((botan_hash_t) hash->handle, (const uint8_t *) buf, len);
+    static_cast<Botan::HashFunction*>(hash->handle)->update((const uint8_t*)buf, len);
```

Comment:
May as well use `static_cast` for `buf` since we're doing C++-style casting here.

Also, what happens when `hash == nullptr`, `buf == nullptr`, `len == 0`?